src/dc_imex.rs

```diff
@@ -998,34 +999,26 @@ unsafe fn export_self_keys(context: &Context, dir: *const libc::c_char) -> bool
 unsafe fn export_key_to_asc_file(
     context: &Context,
     dir: *const libc::c_char,
-    id: libc::c_int,
+    id: Option<i64>,
     key: &Key,
-    is_default: libc::c_int,
 ) -> bool {
     let mut success = false;
-    let dir = as_path(dir);
-
-    let file_name = if 0 != is_default {
-        let name = format!(
-            "{}-key-default.asc",
-            if key.is_public() { "public" } else { "private" },
-        );
-        dir.join(name)
-    } else {
-        let name = format!(
-            "{}-key-{}.asc",
-            if key.is_public() { "public" } else { "private" },
-            id
-        );
-        dir.join(name)
+    let file_name = {
+        let kind = if key.is_public() { "public" } else { "private" };
+        let dir = as_str(dir);
```

Comment:
please use `as_path(dir)` and then `.join()` instead of inserting `/` manually, to avoid issues on non unix platforms